keras_cv/layers/object_detection/non_max_suppression.py

```diff
@@ -112,10 +107,9 @@ def call(self, predictions, images=None):
         )
 
         # preparing the predictions for TF NMS op
-        boxes = tf.expand_dims(predictions[..., :4], axis=2)
-        class_predictions = tf.cast(predictions[..., 4], tf.int32)
-        scores = predictions[..., 5]
-
+        boxes = tf.expand_dims(predictions["boxes"], axis=2)
+        class_predictions = tf.cast(predictions["classes"], tf.int32)
+        scores = predictions["confidence"]
```

Comment:
Just return whatever is returned from tf.image.combined_nms without converting to ragged and TensorArray operations